src/SS2ERC721.sol

```diff
@@ -113,14 +113,19 @@ abstract contract SS2ERC721 is ERC721 {
         require(owner != address(0), "NOT_MINTED");
     }
 
-    function balanceOf(address owner) public view virtual override returns (uint256) {
+    function balanceOf(address owner) public view virtual override returns (uint256 balance) {
         require(owner != address(0), "ZERO_ADDRESS");
 
-        int256 balance = int256(_balanceOfPrimary(owner)) + _balanceOfAdjustment[owner];
-
-        require(balance >= 0, "OVERFLOW");
-
-        return uint256(balance);
+        // if balanceOfAdjustment < 0 then it means the primary owner has transferred out their token
+        // and we can safely return 0 which the named return is already initialized to
+        int256 balanceOfAdjustment = _balanceOfAdjustment[owner];
+        if (balanceOfAdjustment >= 0) {
+            unchecked {
+                int256 balanceInt = int256(_balanceOfPrimary(owner)) + balanceOfAdjustment;
+                require(balanceInt >= 0, "OVERFLOW");
```

Comment:
I'm not sure this check is helping much.  If we feel comfortable that `_balanceOfPrimary` will only ever be 0 or 1, then the only case this require protects us from is when `balanceOfAdjustments` is at type(int).max.

If we do away  with this check then we can directly cast the above into a uint for the return and save more gas...

Reply:
yes the check is a little kludgy. I can sort of convince myself that we would be safe without it, but I'd like to have better tests in place to raise confidence that we can remove it